provider/grpc.go

```diff
@@ -205,48 +205,36 @@ func (g *GRPCClient) SubmitTradeSwap(ctx context.Context, ownerAddress, inToken,
 		Project:      project,
 	})
 	if err != nil {
-		return nil, err
+		errors = append(errors, err)
+		return signatures, errors
 	}
 
-	var signatures []string
 	for _, tx := range resp.Transactions {
 		signature, err := g.signAndSubmit(ctx, tx, skipPreFlight)
-		if err != nil {
-			if signature != "" {
-				signatures = append(signatures, signature)
-			}
-			return signatures, err
-		}
-
 		signatures = append(signatures, signature)
+		errors = append(errors, err)
```

Comment:
Perhaps we can write a common function for submit transactions, with some options? Whatever the case we'll have to write some documentation on this.

```go

type SubmitStrategy int

const (
    SubmitStrategyAll SubmitStrategy = iota  // submit all and collect signatures
    SubmitStrategyDeferLast                            // always execute last transaction even if there's an error
    SubmitStrategyAbortOnError                      // stop on submitting if simulation error
    SubmitStrategyWaitConfirmation               // wait for full confirmation of previous transaction before moving to next, abort if error
)
```

We could also extend the submission endpoint to accept multiple transactions?